src/Service/CaptureService.php

```diff
@@ -339,6 +246,93 @@ public function handleCaptureNotification(
         }
     }
 
+    /**
+     * @param OrderTransactionEntity $orderTransaction
+     * @param NotificationEntity $notification
+     * @return bool
+     */
+    public function checkRequiredAmountFullyCapturedInNotification(
+        OrderTransactionEntity $orderTransaction,
+        NotificationEntity $notification
+    ):bool {
+        return $notification->getAmountValue() >= $this->getRequiredCaptureAmount($orderTransaction->getOrderId());
+    }
+
+    /**
+     * @param OrderTransactionEntity $orderTransaction
+     * @return bool
+     */
+    public function isRequiredAmountCaptured(OrderTransactionEntity $orderTransaction): bool
+    {
+        $requiredCaptureAmount = $this->getRequiredCaptureAmount($orderTransaction->getOrderId());
+        $totalCapturedAmount = $this->getTotalCapturedAmount($orderTransaction);
+
+        return $totalCapturedAmount >= $requiredCaptureAmount;
+    }
+
+    /**
+     * @param string $orderId
+     * @return int
+     */
+    public function getRequiredCaptureAmount(string $orderId): int
+    {
+        $adyenPayments = $this->adyenPaymentService->getAdyenPayments($orderId);
+        $requiredCaptureAmount = 0;
+
+        /** @var AdyenPaymentEntity $adyenPayment */
+        foreach ($adyenPayments as $adyenPayment) {
+            if ($adyenPayment->getCaptureMode() === AdyenPaymentService::MANUAL_CAPTURE) {
+                $requiredCaptureAmount += $adyenPayment->getAmountValue();
+            }
+        }
+
+        return $requiredCaptureAmount;
+    }
+
+    private function getLineItemsArray(
+        ?OrderLineItemCollection $lineItems,
+        string $currencyCode
+    ): array {
+        $lineItemsArray = [];
+        $lineIndex = 0;
+        foreach ($lineItems as $lineItem) {
+            // Skip non-product line items.
+            if (empty($lineItem->getProductId()) || $lineItem->getType() !== LineItem::PRODUCT_LINE_ITEM_TYPE) {
```

Comment:
We could add support for other line items types, same as in #343

Reply:
Thank you for the review @peterojo! I pushed the changes.